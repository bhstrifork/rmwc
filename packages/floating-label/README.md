# floating-label

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test floating-label` to execute the unit tests via [Vitest](https://vitest.dev/).
