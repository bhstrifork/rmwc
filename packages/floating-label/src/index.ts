export * from './lib/floating-label';
