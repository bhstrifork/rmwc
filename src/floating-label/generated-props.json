{
  "npm": {},
  "typescript": {
    "useFloatingLabelFoundation": {
      "fileName": "src/floating-label/foundation.tsx",
      "flags": {
        "isExternal": false,
        "isOptional": false,
        "isPrivate": false,
        "isProtected": false,
        "isPublic": false,
        "isRest": false,
        "isStatic": false
      },
      "kind": "method",
      "name": "useFloatingLabelFoundation",
      "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/floating-label/foundation.tsx#L11",
      "signatures": [
        {
          "fileName": "src/floating-label/foundation.tsx",
          "kind": "signature",
          "name": "useFloatingLabelFoundation",
          "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/floating-label/foundation.tsx#L11",
          "parameters": [
            {
              "flags": {
                "isExternal": false,
                "isOptional": false,
                "isPrivate": false,
                "isProtected": false,
                "isPublic": false,
                "isRest": false,
                "isStatic": false
              },
              "kind": "parameter",
              "name": "props",
              "type": "FloatingLabelProps & HTMLProps<any>"
            }
          ],
          "returnType": "any",
          "type": "(props: FloatingLabelProps & HTMLProps<any>) => any"
        }
      ]
    },
    "FloatingLabelProps": {
      "fileName": "src/floating-label/index.tsx",
      "flags": {
        "isExternal": false,
        "isOptional": false,
        "isPrivate": false,
        "isProtected": false,
        "isPublic": false,
        "isRest": false,
        "isStatic": false
      },
      "kind": "interface",
      "name": "FloatingLabelProps",
      "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/floating-label/index.tsx#L7",
      "methods": [],
      "properties": [
        {
          "fileName": "src/floating-label/index.tsx",
          "flags": {
            "isExternal": false,
            "isOptional": true,
            "isPrivate": false,
            "isProtected": false,
            "isPublic": false,
            "isRest": false,
            "isStatic": false
          },
          "kind": "property",
          "name": "float",
          "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/floating-label/index.tsx#L9",
          "type": "undefined | false | true"
        },
        {
          "fileName": "src/floating-label/index.tsx",
          "flags": {
            "isExternal": false,
            "isOptional": true,
            "isPrivate": false,
            "isProtected": false,
            "isPublic": false,
            "isRest": false,
            "isStatic": false
          },
          "kind": "property",
          "name": "shake",
          "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/floating-label/index.tsx#L8",
          "type": "undefined | false | true"
        }
      ]
    },
    "FloatingLabelApi": {
      "fileName": "src/floating-label/index.tsx",
      "flags": {
        "isExternal": false,
        "isOptional": false,
        "isPrivate": false,
        "isProtected": false,
        "isPublic": false,
        "isRest": false,
        "isStatic": false
      },
      "kind": "interface",
      "name": "FloatingLabelApi",
      "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/floating-label/index.tsx#L12",
      "methods": [],
      "properties": [
        {
          "fileName": "src/floating-label/index.tsx",
          "flags": {
            "isExternal": false,
            "isOptional": false,
            "isPrivate": false,
            "isProtected": false,
            "isPublic": false,
            "isRest": false,
            "isStatic": false
          },
          "kind": "property",
          "name": "getWidth",
          "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/floating-label/index.tsx#L13",
          "type": "() => number"
        },
        {
          "fileName": "src/floating-label/index.tsx",
          "flags": {
            "isExternal": false,
            "isOptional": false,
            "isPrivate": false,
            "isProtected": false,
            "isPublic": false,
            "isRest": false,
            "isStatic": false
          },
          "kind": "property",
          "name": "setRequired",
          "sourceUrl": "https://github.com/rmwc/rmwc/blob/master/src/floating-label/index.tsx#L14",
          "type": "(isRequired: boolean) => void"
        }
      ]
    }
  }
}
