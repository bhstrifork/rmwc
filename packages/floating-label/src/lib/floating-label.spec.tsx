import { render } from '@testing-library/react';
import React from 'react';
import { FloatingLabel } from './floating-label';

describe('FloatingLabel', () => {
  it('renders', () => {
    const { asFragment } = render(<FloatingLabel />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('renders with float', () => {
    const { asFragment } = render(<FloatingLabel float />);
    expect(asFragment()).toMatchSnapshot();
  });
});
