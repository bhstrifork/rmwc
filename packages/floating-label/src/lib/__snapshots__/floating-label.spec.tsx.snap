// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FloatingLabel > renders 1`] = `
<DocumentFragment>
  <span
    class="mdc-floating-label"
  />
</DocumentFragment>
`;

exports[`FloatingLabel > renders with float 1`] = `
<DocumentFragment>
  <span
    class="mdc-floating-label mdc-floating-label--float-above"
  />
</DocumentFragment>
`;
